### PR TITLE
Macos

### LINKED ISSUE #862: Support MacOS client
Support MacOS as a client

### PR DESCRIPTION
**Pull request by ghoneycutt**
# Pull Request Checklist

Add client support for MacOS

## Description

Add client support for MacOS

## Related Issue
Rebase of Issue #826 

Fixes #862

## Motivation and Context

Supports another platform

## How Has This Been Tested?

- [x] Vagrant

## General

- [x] Update `README.md` with any necessary configuration snippets

- [x] New parameters are documented (no new params)

- [x] New parameters have tests (no new params)

- [x] New tests are written

- [x] Tested in Vagrant with Sensu Enterprise

- [x] csoleimani has signed off on this

### CODE REVIEW

File: manifests/server/service.pp [hunk 1]
```diff
@@ -21,7 +21,7 @@
       }
     }
 
-    if $::osfamily != 'windows' {
+    if $::osfamily !~ /(windows|Darwin)/ {
```
Comment (ghoneycutt) [inline manifests/server/service.pp]: instead of a negative regex this should just be

```puppet
if $::kernel == 'Linux' {
```

**Comment by ghoneycutt** [comment]
@alvagante This is working in vagrant, though needs spec tests added for all the new changes. Could you please have a look.

**Comment by ghoneycutt** [comment]
Needs to be tested with sensu-enterprise

**Comment by ghoneycutt** [comment]
@csoleimani can you please test this and let us know how it goes?

**Comment by csoleimani** [comment]
yup, will test it with our environment tomorrow

**Comment by csoleimani** [comment]
seems to work fine for me 👍

**Comment by ghoneycutt** [comment]
Thanks @csoleimani and @alvagante for all the hard work to add MacOS support!!

**Comment by ghoneycutt** [comment]
Released in v2.45.0

**Pull request merged**